ytsaurus_models.go

```diff
@@ -4,25 +4,37 @@ import (
 	"time"
 )
 
+type SourceUser interface {
+	GetID() ObjectID
+	GetName() string
+	GetSourceType() SourceType
+	GetRaw() (map[string]any, error)
+}
+
+type SourceGroup interface {
+	GetID() ObjectID
+	GetName() string
+	GetSourceType() SourceType
+	GetRaw() (map[string]any, error)
+}
+
+type SourceGroupWithMembers struct {
+	SourceGroup SourceGroup
+	// Members is a set of strings, representing users' ObjectID.
+	Members StringSet
+}
+
 type YtsaurusUser struct {
 	// Username is a unique @name attribute of a user.
-	Username string
-	// AzureID is non-human readable string like 2cd8a70c-9044-4488-b06a-c8461c39b296.
-	AzureID string
-	// PrincipalName is a unique human-readable login.
-	// It could be in form of email, but doesn't give guarantee that such email exists.
-	PrincipalName string
-	// Email is filled if Azure user has email.
-	Email       string
-	FirstName   string
-	LastName    string
-	DisplayName string
+	Username    string
+	SourceType  *string
+	SourceRaw   map[string]any
 	BannedSince time.Time
 }
 
 // IsManuallyManaged true if user doesn't have @azure attribute (system or manually created user).
-func (u YtsaurusUser) IsManuallyManaged() bool {
-	return u.AzureID == ""
+func (u YtsaurusUser) IsManuallyManaged(sourceType SourceType) bool {
+	return u.SourceRaw == nil || u.SourceType == nil || *u.SourceType != string(sourceType)
 }
```

Comment:
Basically managed user must have non empty/non nil `SourceRaw`, lets just add a `IsManuallyManaged` method to `YtsaurusUser`, that would check that emptiness of this map. And so we don't accidentally initialise this structure wrongly lets add:
- `NewAutoManagedYtsaurusUser(username, sourceRaw, bannedSince)` function, that would return error if function is called with empty `sourceRaw`.
- `NewManualManagedYtsaurusUser(username)` for creating manually managed users.